pints/_log_likelihoods.py

```diff
@@ -117,6 +117,120 @@ def __call__(self, x):
                       - np.sum(autocorr_error**2, axis=0) / (2 * sigma**2))
 
 
+class GaussianIntegratedUniformLogLikelihood(pints.ProblemLogLikelihood):
+    """
+    Calculates a log-likelihood assuming independent Gaussian-distributed noise
+    at each time point where it is assumed that the dependence on
+    :math:`\\sigma\\sim U(a,b)` has been integrated out of the joint posterior.
```

Comment:
What's a joint posterior?

Reply:
@MichaelClerx See above comments and new docstrings/example notebook.